asdf_astropy/converters/coordinates/spectral_coord.py

```diff
@@ -0,0 +1,39 @@
+
+from asdf.extension import Converter
+from asdf.tags.core.ndarray import NDArrayType
+
+
+class SpectralCoordConverter(Converter):
+    tags = ["tag:astropy.org:astropy/coordinates/spectralcoord-*"]
+
+    types = ["astropy.coordinates.spectral_coordinate.SpectralCoord"]
+
+    def to_yaml_tree(self, obj, tag, ctx):
+        node = {
+            "value": obj.value,
+            "unit": obj.unit,
+        }
+
+        if obj.observer is not None:
+            node["observer"] = obj.observer
+
+        if obj.target is not None:
+            node["target"] = obj.target
+
+        return node
+
+    def from_yaml_tree(self, node, tag, ctx):
+        from astropy.coordinates.spectral_coordinate import SpectralCoord
+
+        value = node["value"]
+        if isinstance(value, NDArrayType):
```

Comment:
For some reason the original astropy tag class was careful not to allow NDArrayType:

https://github.com/astropy/astropy/blob/main/astropy/io/misc/asdf/tags/unit/quantity.py#L33

I don't know why that would be necessary, possibly because someone is doing an `isinstance(value, np.ndarray)` somewhere which fails for NDArrayType.

Reply:
Maybe we should not allow an `NDArrayType` in the future too!  😆 

It is odd that for everything else deserialize the actual object, but not for `np.ndarray`.  I'm sure there's _reasons_, but it is a very weird implementation.

Reply:
It generally seems to cause more trouble than it's worth.